src/circuit/gadget/sinsemilla/merkle/chip.rs

```diff
@@ -270,24 +265,24 @@ impl MerkleInstructions<pallas::Affine, MERKLE_DEPTH_ORCHARD, { sinsemilla::K },
         // Check that the pieces have been decomposed properly.
         /*
             The pieces and subpieces are arranged in the following configuration:
-            |  A_0  |  A_1  |  A_2  |  A_3  |  A_4  | l_plus_1 |
-            ----------------------------------------------------
-            |   a   |   b   |   c   |  left | right |   l + 1  |
-            |  z1_a |  z1_b |  b_1  |  b_2  |       |          |
+            |  A_0  |  A_1  |  A_2  |  A_3  |  A_4  | q_decompose |
+            -------------------------------------------------------
+            |   a   |   b   |   c   |  left | right |      1      |
+            |  z1_a |  z1_b |  b_1  |  b_2  |   l   |             |
         */
         {
             layouter.assign_region(
                 || "Check piece decomposition",
                 |mut region| {
-                    // Set the fixed column `l_plus_1` to the current l + 1.
+                    // Set the fixed column `l` to the current l.
                     // Recall that l = MERKLE_DEPTH_ORCHARD - layer - 1.
                     // The layer with 2^n nodes is called "layer n".
-                    let l_plus_1 = (l as u64) + 1;
-                    region.assign_fixed(
-                        || format!("l_plus_1 {}", l_plus_1),
-                        config.l_plus_1,
-                        0,
-                        || Ok(pallas::Base::from_u64(l_plus_1)),
+                    config.q_decompose.enable(&mut region, 0)?;
+                    region.assign_advice_from_constant(
```

Comment:
Given that we had the spare advice cell, I agree with this change. We could have reused a fixed column if necessary (these regions end up layed out in areas where we have plenty of free space), but having a region depend on fewer columns is better in general for layout optimizations, and equality constraints between columns that are already in the global permutation (which is the case for all advice columns and the constants-enabled column) are essentially free.